zio-http/src/main/scala/zio/http/package.scala

```diff
@@ -1,17 +1,20 @@
 package zio
 
-import zio.http.socket.WebSocketChannelEvent
 import zio.stacktracer.TracingImplicits.disableAutoTrace // scalafix:ok;
 
 package object http extends PathSyntax with RequestSyntax with RouteDecoderModule {
-  type HttpApp[-R, +E] = Http[R, E, Request, Response]
-  type UHttpApp        = HttpApp[Any, Nothing]
-  type RHttpApp[-R]    = HttpApp[R, Throwable]
-  type EHttpApp        = HttpApp[Any, Throwable]
-  type UHttp[-A, +B]   = Http[Any, Nothing, A, B]
 
-  type ResponseZIO[-R, +E]                   = ZIO[R, E, Response]
-  type UMiddleware[+AIn, -BIn, -AOut, +BOut] = Middleware[Any, Nothing, AIn, BIn, AOut, BOut]
+  type RequestHandler[-R, +Err] = Handler[R, Err, Request, Response]
+  type AppHandler[-R]           = RequestHandler[R, Response]
+
+  type RequestHandlerMiddleware[-R, +Err] = HandlerMiddleware[R, Err, Request, Response, Request, Response]
+  type AppHandlerMiddleware[-R]           = RequestHandlerMiddleware[R, Response]
+
+  type HttpRoute[-R, +Err] = Http[R, Err, Request, Response]
```

Comment:
@vigoo Let's not use any `Route` terminology, and let's minimize name changes (so keep `HttpApp` instead of `App`, etc.).

Reply:
Old names for reference:

```scala
  type HttpApp[-R, +E] = Http[R, E, Request, Response]
  type UHttpApp        = HttpApp[Any, Nothing]
  type RHttpApp[-R]    = HttpApp[R, Throwable]
  type EHttpApp        = HttpApp[Any, Throwable]
  type UHttp[-A, +B]   = Http[Any, Nothing, A, B]

  type ResponseZIO[-R, +E]                   = ZIO[R, E, Response]
  type UMiddleware[+AIn, -BIn, -AOut, +BOut] = Middleware[Any, Nothing, AIn, BIn, AOut, BOut]
```

Reply:
so the old `HttpApp` is what currently is `HttpRoute`. If that's `HttpApp` we still need to call somehow the "final" thing, which fixes the `Err` to `Response` - that's what I called `App[R]`. Do you have a better idea?

Reply:
Maybe we just don't need type aliases for that and use `HttpApp[R, Response]` in the code.